src/lib/shadow-shim-helper-rs/src/util/mod.rs

```diff
@@ -0,0 +1,20 @@
+/// A trait to prevent type inference during function calls. Useful when you have a type that wraps
+/// a pointer (like `ForeignArrayPtr`) and you don't want Rust to infer the type of pointer
+/// during creation.  Instead, the caller must specify the generic type.
+///
+/// Example:
+///
+/// ```ignore
+/// let x: ForeignArrayPtr<u8>;
```

Comment:
`let mut x` ?

Reply:
I don't think the `mut` is needed here since we assign it below but don't modify it. For example it's similar to

```rust
let x;
x = 0u32;
```